### PR TITLE
Add test cases to cover all PDO data types and APIs.

### PR DESCRIPTION
**Pull request by SimbaGithub**
1. add test cases to cover all pdo data types and APIs.
2. remove some test cases for timestamp that have different result on different platforms.
3. add test environment setup script for Windows.

The test instruction for Linux can apply to MacOS as well.
To run test on Windows:
1 make sure PHP_HOME is set in environment variable
2 under pdo_snowflake folder, run .\scripts\env.bat
3 use the test script provided by php sdk to run test, for an example:
%PHP_HPME%\php.exe c:\php-sdk\phpmaster\vc15\x64\php-src\run-test.php c:\pdo_snowflake\tests

### CODE REVIEW

File: tests/selectdate.phpt [hunk 1]
```diff
@@ -27,12 +27,12 @@ pdo_snowflake.cacert=libsnowflakeclient/cacert.pem
...
-            "input" => [3, "0001-01-01"],
-            "output" => [3, "1-01-01"],
+            "input" => [3, "1001-01-01"],
+            "output" => [3, "1001-01-01"],
         ],
         [
-            "input" => [4, "0000-01-01"],
-            "output" => [4, "0-01-01"],
+            "input" => [4, "1000-01-01"],
+            "output" => [4, "1000-01-01"],
```
Comment (sfc-gh-kwagner) [inline tests/selectdate.phpt:35]: Why were these test cases changed here?

File: tests/selectntz.phpt [hunk 2]
```diff
@@ -31,12 +31,12 @@ pdo_snowflake.cacert=libsnowflakeclient/cacert.pem
...
-            "input" => [4, "0001-01-01 00:00:00.000"],
-            "output" => [4, "1-01-01 00:00:00.000000000"],
+            "input" => [4, "1001-01-01 00:00:00.000"],
+            "output" => [4, "1001-01-01 00:00:00.000000000"],
         ],
         [
-            "input" => [5, "0000-01-01 00:00:00.000"],
-            "output" => [5, "0-01-01 00:00:00.000000000"],
+            "input" => [5, "1000-01-01 00:00:00.000"],
+            "output" => [5, "1000-01-01 00:00:00.000000000"],
```
Comment (sfc-gh-kwagner) [inline tests/selectntz.phpt:39]: Same here

**Review by sfc-gh-kwagner — COMMENTED** [review]
Looks good, but wondering why those time test cases were changed

File: tests/selectdate.phpt (hunk 1 above)
Comment (SimbaGithub) [reply tests/selectdate.phpt:35]: That's for removing inconsistent test result.
On Windows and Mac the output of 0001-01-01 is 0001-01-01, while on Linux it's 1-01-01. So the test case failed. I believe that's caused by the different behavior of strftime() on %Y, and the result looks acceptable to me.

Comment (sfc-gh-kwagner) [reply tests/selectdate.phpt:35]: Can we keep this as a test case just for Linux then? We originally wrote this test to make sure the driver would be have correctly in this situation where we enter 0001 for the year and the strftime() function on Linux returns 1 as the year

Comment (SimbaGithub) [reply tests/selectdate.phpt]: Updated. Learned how to use EXPECTF.

**Review by sfc-gh-kwagner — APPROVED** [review]

**Pull request merged**